### PR TITLE
emboss: properly sanitize regex

### PR DESCRIPTION
**Pull request by bernt-matthias**
fixes: https://github.com/galaxyproject/tools-iuc/issues/3060

also add tests

FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/emboss_5/macros.xml [hunk 1]
```diff
@@ -19,4 +19,27 @@
...
+          <add value="^"/>
+          <add value="$"/>
+          <add value="("/>
+          <add value=")"/>
+          <add value="|"/>
+          <add value="?"/>
+          <add value="*"/>
+          <add value="+"/>
+	  <add value="{"/>
+	  <add value="}"/>
```
Comment (nsoranzo) [inline tools/emboss_5/macros.xml:34]: Indentation contains tabs

File: tools/emboss_5/macros.xml [hunk 2]
```diff
@@ -19,4 +19,27 @@
...
+          <add value="{"/>
+          <add value="}"/>
+          <add value="\"/>
+          <add value="["/>
+          <add value="]"/>
+          <add value="."/>
+          <add value=","/>
+        </valid>
+      </sanitizer>
+      <validator type="regex" message="Pattern must not be empty.">^.*$</validator>
```
Comment (nsoranzo) [inline tools/emboss_5/macros.xml:42]:
```suggestion
      <validator type="empty_field" />
```

File: tools/emboss_5/macros.xml [hunk 3]
```diff
@@ -19,4 +19,27 @@
...
+      <validator type="regex" message="Pattern must not end with backslash.">^.*[^\\]$</validator>
```
Comment (nsoranzo) [inline tools/emboss_5/macros.xml:43]:
```suggestion
      <validator type="regex" message="Pattern must not end with backslash.">.*[^\\]$</validator>
```

File: tools/emboss_5/macros.xml [hunk 4]
```diff
@@ -19,4 +19,27 @@
             <yield />
         </citations>
     </xml>
+    <xml name="regex_sanitizer">
+      <sanitizer>
```
Comment (nsoranzo) [inline tools/emboss_5/macros.xml:23]: Use 4-spaces indent?

**Review by wm75 — COMMENTED** [review]
Don't forget to bump the wrapper versions!

**Comment by bernt-matthias** [comment]
> Don't forget to bump the wrapper versions!

Thanks. Done.

**Review by wm75 — APPROVED** [review]

**Pull request merged**